### PR TITLE
Made Makefile cross-platform (Linux, Mac)

### PR DESCRIPTION
**Pull request by daveyproctor**
Indeed, thanks to @crossroads1112 for step by step guidance and @americanhanko for his first pass in [PR 86](https://github.com/cs50/libcs50/pull/86)

### CODE REVIEW

**Comment by crossroads1112** [comment]
I looked this over with Davey before he submitted the PR. Looks good to me!

File: Makefile [hunk 1]
```diff
@@ -1,40 +1,63 @@
...
+	cp -r $(SRC) $(INCLUDE) build/src/libcs50
 
 .PHONY: install
-install: build docs
-	mkdir -p $(DESTDIR) $(DESTDIR)/share/man/man3
+install: all
+	mkdir -p $(DESTDIR) $(DESTDIR)/$(MANDIR)
 	cp -r build/* $(DESTDIR)
-	cp -r debian/docs/* $(DESTDIR)/share/man/man3
+	cp debian/docs/* $(DESTDIR)/$(MANDIR)
```
Comment (kzidane) [inline Makefile]: @crossroads1112 @daveyproctor is `ldconfig` needed for Mac?

**Comment by americanhanko** [comment]
Ran into this issue during my first attempt. Is this expected?

**Comment by daveyproctor** [comment]
@americanhanko same issue with `sudo make install`?

File: Makefile (hunk 1 above)
Comment (daveyproctor) [inline Makefile]: @kzidane are you asking about the Linux `export LD_LIBRARY_PATH=/usr/local/lib` command asked for on line 35 of the updated README.md? 

As per [Apple Dynamic Libraries](https://developer.apple.com/library/content/documentation/DeveloperTools/Conceptual/DynamicLibraries/100-Articles/UsingDynamicLibraries.html) documentation, `/usr/local` is automatically checked for libraries. 

A user can install elsewhere as desired, setting `LD_LIBRARY_PATH`.

**Comment by daveyproctor** [comment]
@kzidane ideas on how to pass travis-ci tests?

File: Makefile (hunk 1 above)
Comment (kzidane) [inline Makefile]: Sounds good! Mind running `ldconfig` if installing on Linux, please?

**Comment by kzidane** [comment]
@daveyproctor don't worry about that! It's complaining because you're PR'ing from a fork, and we have some environment variables set that are not available when building PRs from forks for security reasons. If you have write-access to this repo, you may push a branch and PR from that. Otherwise, I can fix later after merging, instead of having you worry about modifying the `.travis.yml` file. How does that sound?

**Comment by crossroads1112** [comment]
:shipit:

**Comment by daveyproctor** [comment]
@dmalan ready to merge?

**Comment by kzidane** [comment]
@dmalan any takes on this one?

**Comment by americanhanko** [comment]
@daveyproctor Apologies for the delay in my response! `sudo make install` works on macOS and files that include `cs50.h` compile successfully with both `gcc` and `clang`. As expected, it does set `root` as the owner of`/usr/local/src/`. I don't foresee any issues with that though unless something else were to utilize that directory with different permissions. 

Given all that, I do think a more elegant solution to installation on macOS would be to create a [Homebrew](https://brew.sh/) recipe for `libcs50` that utilizes the changes in this PR. This way, a student using macOS could `install`, `uninstall` or `update` with `brew`. Any thoughts on that?

Either way, thanks! Great work on this.

**Comment by daveyproctor** [comment]
@americanhanko I did a little research on this, per the below. Unless I'm missing something(?), a problem with Brew seems to be that even upping the version number requires a PR. Rather unergonomic, if so. My full review:

# Getting libcs50 via Brew

## Goal

```
$ brew install libcs50
==> Using the sandbox
==> Downloading https://github.com/cs50/libcs50/archive/v8.0.5.zip
######################################################################## 100.0%
==> make install
🍺  /usr/local/Cellar/libcs50/8.0.5: 14 files, 33.2KB, built in 2 seconds
```

## Steps and UX
* Create a brew formula [how to: Formula Cookbook](https://github.com/Homebrew/brew/blob/master/docs/Formula-Cookbook.md)
    * Everything happens via Github.
    * Fork the homebrew-core repo, add a Ruby "formula", and PR.
    * [example formulas](https://github.com/Homebrew/homebrew-core/tree/master/Formula)
    * `$brew create https://github.com/cs50/libcs50/archive/v8.0.5.zip` opens a pre-populated file `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libcs50.rb`
    * Formula contains
        * a link to source zip / tarball
        * Minimum 1 test to make sure library is installed properly
* On `brew install cs50`, Brew
    * gets and extracts the library
    * runs `./configure` and `make install`, by default
        * installed into `/usr/local/Cellar/`
        * symlinks `/usr/local/*` to the Cellar
    * runs tests specified in the Formula
* On `brew upgrade cs50`, Brew
    * checks the current version against the formula in GitHub
    * reinstalls as needed

## Concerns + workarounds
* As they say in the Formula Cookbook, upping the library version involves creating a new pull request with a trivially-changed brew formula.
    * [Example](https://github.com/Homebrew/homebrew-core/pull/15761/files).
    * This is bad because we would be reliant on brew in order to keep our libary versions in sync via multiple download methods.
    * Workaround
        * Brew makes this somewhat ergonomic with the command `brew bump-formula-pr`, which forks, commits, and pushes all in one step. [Link](http://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request.html)
            * Also, `brew bump-formula-pr --help`
* Installing into `/usr/local/Cellar` and symlinking `/usr/local/*` to it is different from installing from source on Mac and Linux, as per [PR 97](https://github.com/cs50/libcs50/pull/97)
    * Workaround
        * We could keep a different Makefile, `Makefile.brew` for Brew to use. In our formula, we could do `make -f Makefile.brew`

**Comment by americanhanko** [comment]
@daveyproctor thanks for the quick reply (and the excellent review).

> We could keep a different Makefile, Makefile.brew for Brew to use. In our formula, we could do make -f Makefile.brew

This is one of the main reasons I chose to create `Makefile.macos` in https://github.com/cs50/libcs50/pull/86. If keeping a different Makefile for Homebrew is legitimately an option, feel free to take from the one I wrote. Also, I completely agree with naming it `Makefile.brew` - it would better communicate its purpose.

Perhaps another option would be to write another target definition in the Makefile (e.g. `make brew`) which could be called in the Homebrew formula.

**Comment by crossroads1112** [comment]
Homebrew concerns aside, I think we should merge this PR so we get rudimentary support for OSX and support for making as non-root on Linux. We can worry about Homebrew after that.

**Comment by daveyproctor** [comment]
@crossroads1112 would you consider pre-building the man pages on our end so the end user doesn't need to worry about asciidoctor?

**Comment by crossroads1112** [comment]
So every time a change is made to the `adoc` files, we'd have to remember to regenerate the man pages? That doesn't feel super clean to me but maybe there's a better way? In any case, I don't really have a problem with keeping asciidoctor as a dependency.

**Comment by daveyproctor** [comment]
That CS50 generates the man pages from `.adoc` seems like a convenience for us more than being relevant to the user. For instance, a maker of a library could write straight to their raw man pages, right? It almost seems like the `.adoc` files should live in our `.gitignore`, and not come with the library at all (only the man pages). Do you happen to know if asciidoctor or other man page generator is often a dependency for library installs?

**Comment by americanhanko** [comment]
@crossroads1112 

> So every time a change is made to the adoc files, we'd have to remember to regenerate the man pages?

Don't you have to do that anyway since the Debian package is distributed with the compiled `man` pages?

**Comment by crossroads1112** [comment]
It happens every time you run `make deb` I think. In any case, this is
really outside the scope of this PR

**Comment by crossroads1112** [comment]
To clarify my previous comment, it feels very unclean to maintain two
different versions of the man pages that, in between releases, are not
necessarily in sync. Avoiding the `asciidoctor` dependency for the `build`
target doesn't feel like a compelling enough reason to do this,
particularly since only more comfy students will be downloading the library
on there own anyway.

I think this issue merits discussion but should be done in a separate
issue/PR. I think this PR should be merged in the meantime while we figure
this (and Homebrew) out.

Thoughts @dmalan?

**Comment by kzidane** [comment]
Merged as part of #106. Thank you!

**Pull request closed without merging**